samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java

```diff
@@ -1962,17 +1964,17 @@ public void populateFib(ItemDataIfc item, ItemContentsBean bean, Map<Long, Answe
     bean.setFibArray(fibs);
   }
 
-  private static List extractFIBFINTextArray(String alltext)
+  private static ArrayList extractFIBFINTextArray(String alltext, String markers_pair)
   {
-    List texts = new ArrayList();
+    ArrayList texts = new ArrayList();
+    String tmp2="";
 
-    while (alltext.indexOf("{}") > -1)
-    {
-      int alltextLeftIndex = alltext.indexOf("{}");
+    while (alltext.indexOf(markers_pair) > -1) {
```

Comment:
Tested on my local instance. Infinite loop
Should be the same as the original and only change the parameter

Reply:
It should be like this:

`  private static ArrayList extractFIBFINTextArray(String alltext, String markers_pair)
  {
	ArrayList texts = new ArrayList();

	while (alltext.indexOf(markers_pair) > -1) {
	  int alltextLeftIndex = alltext.indexOf(markers_pair);
      //int alltextRightIndex = alltext.indexOf("}");

      String tmp = alltext.substring(0, alltextLeftIndex);
      alltext = alltext.substring(alltextLeftIndex + 2);
      texts.add(tmp);
      // there are no more "{}", exit loop. 
      // why do we this check? will it ever come to here?
      if (alltextLeftIndex == -1)
      {
        break;
      }
    }
	texts.add(alltext);
    return texts;
  }`

Reply:
Codacy doesn't allow to modify a parameter inside its function. I have fixed the bug using a temp variable. Thanks!